test/Microsoft.ML.Benchmarks/TextLoaderBench.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Transforms.Text;
+using Xunit;
+
+namespace Microsoft.ML.Benchmarks
+{
+    [Config(typeof(TrainConfig))]
+    public class TextLoaderBench : BenchmarkBase
+    {
+        private MLContext _mlContext;
+        private IDataView _dataView;
+        private static int _numColumns = 100;
+        private static int _numRows = 3000;
+        private static int _maxWordLength = 15;
+        private static int _numColumnsToGet = 20;
+        private List<TextLoader.Column> _columns;
+
+
+        [GlobalSetup]
+        public void SetupData()
+        {
+            Path.GetTempFileName();
+            _mlContext = new MLContext(seed: 1);
+            var path = Path.GetTempFileName();
+            Console.WriteLine($"Created dataset in temporary file:\n{path}\n");
+            path = RandomFile.CreateRandomFile(path, _numRows, _numColumns, _maxWordLength);
+
+            _columns = new List<TextLoader.Column>();
+            for(int i = 0; i < _numColumns; i++)
+            {
+                _columns.Add(new TextLoader.Column($"Column{i}", DataKind.String, i));
+            }
+
+            var textLoader = _mlContext.Data.CreateTextLoader(new TextLoader.Options()
+            {
+                Columns = _columns.ToArray(),
+                HasHeader = false,
+                Separators = new char[] { ',' },
+                AllowQuoting = true,
+                //ReadMultilines = true,
+                //EscapeChar = '\\',
+            });
```

Comment:
Can we do separate benchmarks for different combination of options? If you think that is going to take time, you can back out this file and let the rest of this PR go through and add the benchmark in a separate PR.

Reply:
As discussed offline, I'll keep the benchmark here but uncomment the ReadMultilines and EscapeChar options to actually test them on the benchmark.